OrbitGl/GraphTrack.cpp

```diff
@@ -9,96 +9,84 @@
-        float point_y = y0 - static_cast<float>((max_ - value) * size_[1] / value_range_);
-        const Color kBlack(0, 0, 0, 255);
-        const Color kWhite(255, 255, 255, 255);
-        DrawLabel(canvas, Vec2(point_x, point_y), std::to_string(value), kBlack, kWhite, text_z);
-      }
-    }
-  }
-
   batcher->AddBox(box, color, shared_from_this());
 
-  if (canvas->GetPickingManager().IsThisElementPicked(this)) {
-    color = Color(255, 255, 255, 255);
-  }
-
-  batcher->AddLine(pos_, Vec2(x1, y0), track_z, color, shared_from_this());
-  batcher->AddLine(Vec2(x1, y1), Vec2(x0, y1), track_z, color, shared_from_this());
-
-  const Color kLineColor(0, 128, 255, 128);
-  const Color kDotColor(0, 128, 255, 255);
-
+  const bool picking = picking_mode != PickingMode::kNone;
   if (!picking) {
-    // Current time window
-    uint64_t min_ns = time_graph_->GetTickFromUs(time_graph_->GetMinTimeUs());
-    uint64_t max_ns = time_graph_->GetTickFromUs(time_graph_->GetMaxTimeUs());
-    double time_range = static_cast<float>(max_ns - min_ns);
+    double time_range = static_cast<double>(max_tick - min_tick);
     if (values_.size() < 2 || time_range == 0) return;
 
-    auto it = values_.upper_bound(min_ns);
+    auto it = values_.upper_bound(min_tick);
     if (it == values_.end()) return;
     if (it != values_.begin()) --it;
     uint64_t previous_time = it->first;
     double last_normalized_value = (it->second - min_) * inv_value_range_;
     constexpr float kDotRadius = 2.f;
     float base_y = pos_[1] - size_[1];
-    DrawSquareDot(canvas,
+    DrawSquareDot(batcher,
                   Vec2(time_graph_->GetWorldFromTick(previous_time),
                        base_y + static_cast<float>(last_normalized_value) * size_[1]),
                   kDotRadius, dot_z, kDotColor);
+
     for (++it; it != values_.end(); ++it) {
-      if (previous_time > max_ns) break;
+      if (previous_time > max_tick) break;
       uint64_t time = it->first;
       double normalized_value = (it->second - min_) * inv_value_range_;
       float x0 = time_graph_->GetWorldFromTick(previous_time);
       float x1 = time_graph_->GetWorldFromTick(time);
       float y0 = base_y + static_cast<float>(last_normalized_value) * size_[1];
       float y1 = base_y + static_cast<float>(normalized_value) * size_[1];
-      time_graph_->GetBatcher().AddLine(Vec2(x0, y0), Vec2(x1, y1), graph_z, kLineColor);
-      DrawSquareDot(canvas, Vec2(x1, y1), kDotRadius, dot_z, kDotColor);
+      batcher->AddLine(Vec2(x0, y0), Vec2(x1, y1), graph_z, kLineColor);
+      DrawSquareDot(batcher, Vec2(x1, y1), kDotRadius, dot_z, kDotColor);
 
       previous_time = time;
       last_normalized_value = normalized_value;
     }
   }
 }
 
-void GraphTrack::DrawSquareDot(GlCanvas* canvas, Vec2 center, float radius, float z, Color color) {
+void GraphTrack::Draw(GlCanvas* canvas, PickingMode picking_mode, float z_offset) {
+  const bool picking = picking_mode != PickingMode::kNone;
+  if (picking) {
```

Comment:
Please put the condition inside the if (as we now are using it only once)

if (picking_mode != PickingMode::kNone) ...